ocaml/xapi/cli_frontend.ml

```diff
@@ -2066,6 +2066,22 @@ add a mapping of 'path' -> '/tmp', the command line should contain the argument
       implementation=No_fd Cli_operations.net_attach;
       flags=[Hidden];
     };
+    "network-add-purpose",
```

Comment:
Why do we need these at all? You've already updated records.ml so we are able to use `xe network-param-set` and friends.

Reply:
I now see the other comments regarding this. I personally see no value in adding these extra functions...

Reply:
If I remember correctly, Jon said a while ago that having different CLI and XenAPI commands is confusing for users. Also, we need to do error checking in the functions that add network purposes, so I guess we cannot just simply use the autogenerated ones?

Reply:
My understanding is that if we want to have bash autocompletion for network purpose values, then we need a command-line parameter called `network-purpose` (or `purpose`), and for that we need to have these functions.

Otherwise we could just use the autogenerated ones.

So, that's the value... but whether it's worthwhile is an open question.